payments/providers/talpa_ecom.py

```diff
@@ -0,0 +1,445 @@
+        )
+        user_hash = get_customer_hash(order.customer)
+
+        headers = {"user": user_hash}
+
+        payload = {
+            "namespace": self.config.get(VENE_PAYMENTS_TALPA_ECOM_API_NAMESPACE),
+            "user": user_hash,
+        }
+
+        self.payload_add_customer(payload, order)
+        self.payload_add_products(payload, order, language)
+
+        try:
+            r = requests.post(
+                self.url_order_experience_api, json=payload, headers=headers, timeout=60
+            )
+            r.raise_for_status()
+            resp = r.json()
+            return self.handle_initiate_payment(order, resp)
+        except RequestException as e:
+            if e.response is not None and (data := e.response.json()):
+                errors = data.get("errors", [])
+                messages = "\n".join([error.get("message", "") for error in errors])
+                raise RequestValidationFailedError(messages)
+
+            raise ServiceUnavailableError(_("Payment service is unreachable")) from e
+
+    def handle_initiate_payment(self, order, response) -> str:
+        if errors := response.get("errors", []):
+            messages = "\n".join([error.get("message", "") for error in errors])
+            raise RequestValidationFailedError(messages)
+
+        if talpa_order_id := response.get("orderId", None):
+            order.talpa_ecom_id = talpa_order_id
+            order.save(update_fields=["talpa_ecom_id"])
+        else:
+            raise MissingOrderIDError(_("Order did not contain an id"))
+
+        return self.url_checkout.format(
+            talpa_ecom_id=talpa_order_id,
+            user_hash=get_customer_hash(order.customer),
+        )
+
+    def payload_add_products(self, payload: dict, order: Order, language: str):
+        """Attach info of bought products to payload
+
+        Order lines that contain bought products are retrieved through order"""
+        order_lines: [OrderLine] = order.order_lines.all()
+        payload["items"] = []
+
+        # Additional product orders doesn't have berth product
+        if hasattr(order, "product") and order.product:
+            self.payload_add_place_product(payload, order, language)
+
+        for order_line in order_lines:
+            with override(language):
+                product_name = order_line.product.name
+            self.payload_add_to_items(payload, order_line, product_name, [])
+
+        payload["priceNet"] = rounded(order.pretax_price, as_string=True)
+        payload["priceTotal"] = rounded(order.total_price, as_string=True)
+        payload["priceVat"] = str(
+            rounded(order.total_price) - rounded(order.pretax_price)
+        )
+
+    def payload_add_to_items(
+        self, payload: dict, item: OrderIsh, product_name: str, meta: list = None
+    ):
+        product = item.product
+
+        payload["items"].append(
+            {
+                "productId": resolve_product_talpa_ecom_id(product),
```

Comment:
In another PR https://github.com/City-of-Helsinki/berth-reservations/pull/626/files I have also talpa_ecom_accounting_id in the TalpaProductAccounting model. Is that needed actually anywhere? If not, I'll remove it and just leave talpa_ecom_product_id

Reply:
Either way is fine, this was just a placeholder to get the integration working while we figured out how to handle all this id stuff 👍